File: Server/src/Domain/Users/Errors/UserErrors.cs
using Domain.Common;

namespace Domain.Users.Errors;

public static class UserErrors
{
    public static Error AlreadyRegistered(string email) => Error.BadRequest("User.AlreadyRegistered", $"There is a user already registered with email :'{email}'.");
    public static Error NotFound(string email = "unknown") => Error.NotFound("User.NotFound", $"User with email: '{email}', not found or password is wrong.");
    public static Error TokenEpired() => Error.NotAuthorized("User.TokenExpired", $"User's token has expired, please login again.");
}

File: Client/src/features/Users/userErrors.ts
export enum userErrors {
  InvalidEmail = "That doesn't look like an email address.",
  InvalidFirstName = 'First Name must be below 30 chars.',
  InvalidLastName = 'Last Name must be below 30 chars.',
  MissingFirstName = 'Please enter your first name.',
  MissingLastName = 'Please enter your last name.',
  MissingPassword = "Can't login without a password.",
  InvalidPassword = 'Password must include 8-30 chars, with uppercase, lowercase, digit, and special symbol.',
  PasswordsMatch = "Passwords don't match.",
  NotFound = 'User with this email not found or password is wrong.',
  EmailInUse = 'There is a user already registered with this email.',
}
